client/chat/channel.tsx

```diff
@@ -182,140 +178,121 @@ interface UserListProps {
   users: UsersRecord
 }
 
-class UserList extends React.Component<UserListProps> {
-  state = {
-    width: 0,
-    height: 0,
-  }
-  _contentRef = React.createRef<HTMLDivElement>()
-  _listRef = React.createRef<VirtualizedList>()
-
-  componentDidMount() {
-    this.updateDimensions()
-  }
+const UserList = React.memo((props: UserListProps) => {
+  const { active, idle, offline } = props.users
+  const [dimensionsRef, containerRect] = useObservedDimensions()
+  const containerRef = useRef<HTMLElement | null>(null)
+  const listRef = useRef<VariableSizeList | null>(null)
+  const containerCallback = useCallback(
+    (node: HTMLElement | null) => {
+      dimensionsRef(node)
+      containerRef.current = node
+    },
+    [dimensionsRef],
+  )
 
-  componentDidUpdate(prevProps: UserListProps) {
-    if (prevProps.users !== this.props.users) {
-      this._listRef.current?.recomputeRowHeights()
-    }
-  }
+  useEffect(() => {
+    listRef.current?.resetAfterIndex(0, false)
```

Comment:
It seems to me like if this is necessary, this is actually too late to be doing this (and the `false` second parameter here means the cache wouldn't be updated until a follow-on render call or if the UI was scrolled?).

I think the right place to call this would be directly during the render, so that react-window will re-request sizes in its own render

Reply:
I added this because the `Overline`s in the `UserList` would shrink to the size of a `UserListEntry` if a user logged in/out. I initially left out the `false`, but it seems to be working now when I set it to `true`? Not sure what I missed there.

By calling directly during the render, do you mean something like this?
```
  return (
    <UserListContainer ref={containerCallback}>
      {listRef.current?.resetAfterIndex(0)}
      <VariableSizeList
```
If so, I get an error `Warning: Cannot update during an existing state transition`

Reply:
By "directly in render" I just mean at some point in the component function (putting it in the JSX part is weird but probably works the same, although I'm not entirely certain on that). Passing `true` as the second param likely forces a second render on every render, so I think it would re-render over and over again forever? Unsure to be honest. I'll have to explore more, but from looking at their code before, calling it in a `useEffect` doesn't seem like the right thing to me.

Honestly I'd prefer they just let us turn off the caching entirely, as this lookup is likely very cheap for this usecase anyway =/

Reply:
Ahh, I gotchya. Also, looks like the error I was getting was actually from the `true` parameter, and it only appeared once on loading the page.

Reply:
Ah okay, that makes sense I think because the `true` parameter basically means "re-render now", and React probably doesn't like doing that if it's currently rendering :)